packages/node/src/lib/retry.spec.ts

```diff
@@ -0,0 +1,275 @@
+
+    nock(BACKEND_URL)
+      .get(TOPICS_PATH)
+      .reply(408, { message: 'Request Timeout' });
+
+    nock(BACKEND_URL)
+      .get(TOPICS_PATH)
+      .replyWithError(new NetworkError('ECONNRESET'));
+
+    nock(BACKEND_URL)
+      .get(TOPICS_PATH)
+      .replyWithError(new NetworkError('ETIMEDOUT'));
+
+    nock(BACKEND_URL)
+      .get(TOPICS_PATH)
+      .replyWithError(new NetworkError('ECONNREFUSED'));
+
+    nock(BACKEND_URL)
+      .get(TOPICS_PATH)
+      .reply(504, { message: 'Gateway timeout' });
+
+    nock(BACKEND_URL)
+      .get(TOPICS_PATH)
+      .reply(422, { message: 'Unprocessable Content' });
+
+    nock(BACKEND_URL).get(TOPICS_PATH).reply(200, [{}, {}]);
+
+    const novuClient = new Novu('fake-key', {
+      backendUrl: BACKEND_URL,
+      retryConfig: {
+        initialDelay: 0,
+        waitMin: 0.2,
+        waitMax: 0.5,
+        retryMax: 7,
+      },
+    });
+
+    const result = await novuClient.topics.list({});
+    expect(result.status).toEqual(200);
+  });
+
+  describe('defaultRetryCondition function', () => {
+    test.each<[number, string]>(NON_RECOVERABLE_ERRORS)(
+      'should return false when HTTP status is %i',
+      (status) => {
+        const err = new HttpError(status);
+        expect(defaultRetryCondition(err as AxiosError)).toEqual(false);
+      }
+    );
+
+    test.each<number>(RETRYABLE_HTTP_CODES)(
+      'should return true when HTTP status is %i',
+      (status) => {
+        const err = new HttpError(status);
+        expect(defaultRetryCondition(err as AxiosError)).toEqual(true);
+      }
+    );
+
+    it('should return true when HTTP status is 500', () => {
+      const err = new HttpError(500);
+      expect(defaultRetryCondition(err as AxiosError)).toEqual(true);
+    });
+
+    it('should return true when network code is ECONNRESET', () => {
+      const err = new NetworkError('ECONNRESET');
+      expect(defaultRetryCondition(err as AxiosError)).toEqual(true);
+    });
+
+    it('shoud return false on unknown error', () => {
+      const err = new Error('Unexpected error');
+      expect(defaultRetryCondition(err as AxiosError)).toEqual(false);
+    });
+  });
+
+  describe('hasAllEqual helper function', () => {
+    it('should return true when all items are equal', () => {
+      const arr = ['a', 'a', 'a', 'a'];
+      expect(hasAllEqual(arr)).toEqual(true);
+    });
+
+    it('should return false when items are not equal', () => {
+      const arr = ['a', 'b', 'b', 'b'];
+      expect(hasAllEqual(arr)).toEqual(false);
+    });
+  });
+
+  describe('hasUniqueOnly helper function', () => {
+    it('should return true when all items are unique', () => {
+      const arr = ['a', 'b', 'c', 'd'];
+      expect(hasUniqueOnly(arr)).toEqual(true);
+    });
+
+    it('should return false when items are not unique', () => {
+      const arr = ['a', 'a', 'c', 'd'];
+      expect(hasUniqueOnly(arr)).toEqual(false);
+    });
+  });
+});
```

Comment:
Excellent tests 👍

Reply:
hey @michaldziuba03 , just a few [spellcheck issues](https://github.com/novuhq/novu/actions/runs/6927540924/job/18841594745?pr=4462) to resolve then we are good to merge 🙌

Reply:
@rifont done